### PR TITLE
make controlled base gate an instance.

### PR DESCRIPTION
**Pull request by ewinston**
### Summary
Change a controlled gate's `base_gate` method to be an instance of the base gate instead of a class.

### CODE REVIEW

File: qiskit/circuit/add_control.py [hunk 1]
```diff
@@ -126,13 +126,13 @@ def control(operation, num_ctrl_qubits=1, label=None):
     instr = qc.to_instruction()
     if isinstance(operation, controlledgate.ControlledGate):
         new_num_ctrl_qubits = num_ctrl_qubits + operation.num_ctrl_qubits
-        base_name = operation.base_gate_name
+        base_name = operation.base_gate.name
         base_gate = operation.base_gate
-        base_gate_name = operation.base_gate_name
+        base_gate_name = operation.base_gate.name
```
Comment (ajavadia) [inline qiskit/circuit/add_control.py]: we can probably just remove `base_gate_name` now

File: qiskit/circuit/controlledgate.py [hunk 2]
```diff
@@ -49,11 +49,10 @@ def __init__(self, name, num_qubits, params, label=None, num_ctrl_qubits=1,
                 if isinstance(base_gate, ControlledGate):
                     self.base_gate = base_gate.base_gate
                 else:
-                    self.base_gate = base_gate.__class__
-
-                self.base_gate_name = base_gate.name
+                    self.base_gate = base_gate
 
     def __eq__(self, other):
+        #import ipdb;ipdb.set_trace()
```
Comment (ajavadia) [inline qiskit/circuit/controlledgate.py]: remove

File: qiskit/circuit/add_control.py (hunk 1 above)
Comment (ewinston) [reply qiskit/circuit/add_control.py]: yes

File: qiskit/circuit/controlledgate.py (hunk 2 above)
Comment (ewinston) [reply qiskit/circuit/controlledgate.py]: done

**Comment by mtreinish** [comment]
This should have included a release note to document the breaking change. I'll add it as part of https://github.com/Qiskit/qiskit-terra/pull/3798 since we're close to the release

**Pull request merged**